sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -204,15 +234,15 @@ class UnivocityParser(
    * Parses a single CSV string and turns it into either one resulting row or no row (if the
    * the record is malformed).
    */
-  def parse(input: String): InternalRow = doParse(input)
+  def parse(input: String): Seq[InternalRow] = doParse(input)
 
   private val getToken = if (options.columnPruning) {
     (tokens: Array[String], index: Int) => tokens(index)
   } else {
     (tokens: Array[String], index: Int) => tokens(tokenIndexArr(index))
   }
 
-  private def convert(tokens: Array[String]): InternalRow = {
+  private def convert(tokens: Array[String]): Seq[InternalRow] = {
```

Comment:
actually we need `Option[IntervalRow]`?

Reply:
nvm, we need Seq at the end anyway.

Reply:
`def parse(input: String): Seq[InternalRow] = doParse(input)`  seems like it can return multiple records whereas it's always one or none. It's really confusing and I don't think it's a good idea to sacrifice readability here.

Can we use `Option` here and fix `FailureSafeParser`'s `rawParser` signature to, for instance, `rawParser: IN => GenTraversableOnce[InternalRow]`?

Reply:
We can do that but modification of `FailureSafeParser` is slightly orthogonal to the purpose of the PR, and it is not necessary for this changes. Can we do that separately?

Reply:
The required change in `FailureSafeParser` is one liner. I think we should use `Option` here instead of `Seq` ...

Reply:
Yup, It should be fine to do it separately